src/window.mm

```diff
@@ -148,6 +148,40 @@ - (void)notified:(const std::string &)note withData:(const msgpack::object &)upd
         AppDelegate *app = (AppDelegate *)[NSApp delegate];
         [app newWindow];
     }
+    else if (note == "neovim.app.openFile") {
+        NSArray *files = [mMainView showFileOpenDialog];
+        if (files != nil) {
+            std::vector<char *> args;
+
+            /* open in tabs if more than one file was selected */
+            if ([files count] > 1)
+                args.push_back(const_cast<char *>("-p"));
+
+            for (NSURL *url in files) 
+                args.push_back(const_cast<char *>([[url path] UTF8String]));
+
+            AppDelegate *app = (AppDelegate *)[NSApp delegate];
+            [app newWindowWithArgs:args];
+        }
+    }
+    else if (note == "neovim.app.saveFile") {
+        NSURL *file = [mMainView showFileSaveDialog];
+        if (file != nil)
+        {
+            std::stringstream cmd;
+            cmd << "w " << [[file path] UTF8String];
+            mVim->vim_command(cmd.str());
+        }
+    }
+    else if (note == "neovim.app.saveAsFile") {
```

Comment:
Preferably.  if you `:sav myfile.txt` instead of `:w myfile.txt` on an unnamed buffer you will end up with two buffers (the one that you saved and the one that is unnamed. 

Likewise doing `:w myfile.txt` instead of `:sav myfile.txt` on a named buffer will write out that file but  the buffer will not changed to the one that was just written.

Just in case my explanation was confusing. From the doc:
```
:sav[eas][!] [++opt] {file}
                        Save the current buffer under the name {file} and set
                        the filename of the current buffer to {file}.  The
                        previous name is used for the alternate file name.
```

We could just have 1 api call that passes an argument and tells it whether to do a `:sav` or `:w`